packages/synthetix-governance/contracts/modules/ElectionModule.sol

```diff
@@ -0,0 +1,115 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {ElectionModule as BaseElectionModule} from "@synthetixio/core-modules/contracts/modules/ElectionModule.sol";
+import "@synthetixio/core-contracts/contracts/utils/MathUtil.sol";
+import "@synthetixio/core-modules/contracts/interfaces/IElectionModule.sol";
+import "../interfaces/ISynthetixElectionModule.sol";
+import "../submodules/election/DebtShareManager.sol";
+import "../submodules/election/CrossChainDebtShareManager.sol";
+
+/// @title Module for electing a council, represented by a set of NFT holders
+contract ElectionModule is ISynthetixElectionModule, BaseElectionModule, DebtShareManager, CrossChainDebtShareManager {
+    function initializeElectionModule(
+        string memory councilTokenName,
+        string memory councilTokenSymbol,
+        address[] memory firstCouncil,
+        uint8 minimumActiveMembers,
+        uint64 nominationPeriodStartDate,
+        uint64 votingPeriodStartDate,
+        uint64 epochEndDate,
+        address debtShareContract
+    ) external onlyOwner onlyIfNotInitialized {
+        _setDebtShareContract(debtShareContract);
+
+        initializeElectionModule(
+            councilTokenName,
+            councilTokenSymbol,
+            firstCouncil,
+            minimumActiveMembers,
+            nominationPeriodStartDate,
+            votingPeriodStartDate,
+            epochEndDate
+        );
+    }
+
+    function nominate() public override(BaseElectionModule, IElectionModule) onlyInPeriod(ElectionPeriod.Nomination) {
+        _takeDebtShareSnapshotOnFirstNomination();
+
+        super.nominate();
+    }
+
+    // ---------------------------------------
+    // Debt shares
+    // ---------------------------------------
+
+    function setDebtShareContract(address debtShareContract)
```

Comment:
do we really want to call it `debtShareContract`? Maybe something more generic for the name?

Reply:
If you have a suggestion, happy to improve naming here.

Reply:
if we could just call it `setVotingWeightContract` for example

Reply:
I disagree on this. If you look at everything else (contracts, etc) based on "debt shares". The contract that we use to determine voting weight is the ElectionModule itself. It just uses debt share data from v2x. I'm going to resolve this, and ask you to look at other nomenclature in the ElectionModule and, if you want to insist, we could open a new ticket?